modules/dhcp/dhcp_api.rb

```diff
@@ -25,6 +29,11 @@ class Proxy::DhcpApi < ::Sinatra::Base
   end
 
   get "/:network/unused_ip" do
+    ip_array = [params[:network]]
+    ip_array << params[:from] unless params[:from].nil?
+    ip_array << params[:to] unless params[:to].nil?
+    server.validate_supported_address(*ip_array)
+    validate_mac(params[:mac]) unless params[:mac].nil?
```

Comment:
```suggestion
    validate_mac(params[:mac]) if params[:mac]
```

is generally better. In RoR there is `present?` helper for strings, but it is not available here unfortunately.

There are couple of other places where you could reverse this.